### PR TITLE
WIP SMS invitation message

### LINKED ISSUE #41: Sending invitation link over SMS
First iteration of #35 

Just sending the same invitation link over SMS, like already done over email

### PR DESCRIPTION
**Pull request by oliverbarnes**
Closes #41

### CODE REVIEW

File: app/controllers/v2/event_invitations_controller.rb [hunk 1]
```diff
@@ -21,11 +21,32 @@ def create
...
-          email_address: email_address,
-          event: event_invitation.event,
-          person: Person.find_by(email_address: email_address)
-        ).deliver_later
+        person = Person.find_by(email_address: email_address)
+
+        if person
+          send_email_notification(event_invitation, person)
+          send_sms_notification person
+        end
```
Comment (marcoow) [inline app/controllers/v2/event_invitations_controller.rb]: I think we only want to send one or the other.

Comment (oliverbarnes) [inline app/controllers/v2/event_invitations_controller.rb]: hmm, was assuming we'd always send the email notification, and an sms as well if opted. @cromulus @asyp would you please confirm what it should be?

**Comment by oliverbarnes** [comment]
Ready for review, I think. Can use some refactoring, but given the time constraints, something for the future

File: app/controllers/v2/event_invitations_controller.rb [hunk 2]
```diff
@@ -21,14 +21,49 @@ def create
...
     def send_notifications(event_invitation)
       event_invitation.email_addresses_to_array.each do |email_address|
-        EventInvitationMailer.invite(
-          email_address: email_address,
-          event: event_invitation.event,
-          person: Person.find_by(email_address: email_address)
-        ).deliver_later
+        person = Person.find_by(email_address: email_address)
+
+        case person.preferred_contact_method
```
Comment (marcoow) [inline app/controllers/v2/event_invitations_controller.rb]: we should maybe change this to `case person.preferred_contact_method.upcase` to make it more robust

**Comment by marcoow** [comment]
Just one small issues. Apart from that it looks good. You might want to rebase on `development` to fix the tests.

File: app/controllers/v2/event_invitations_controller.rb (hunk 2 above)
Comment (oliverbarnes) [inline app/controllers/v2/event_invitations_controller.rb]: done

**Comment by oliverbarnes** [comment]
Thanks. Looks like the branch is already up to date, though. Looking into why the push build is failing

**Comment by oliverbarnes** [comment]
rebuilding made it pass. going ahead and merging this. 

If this test acts up again, I'll take a closer look. I have the impression variations on datetimes each time the builds run are making the slot overlap tests fail.

**Pull request merged**